lib/SILOptimizer/Utils/Generics.cpp

```diff
@@ -1219,7 +1152,7 @@ void ReabstractionInfo::specializeConcreteAndGenericSubstitutions(
         M,
         [&](SubstitutableType *type) -> Type {
           auto SpecializedInterfaceTy =
-              CalleeInterfaceToSpecializedInterfaceType(type);
+              MapCalleeInterfaceTypeToSpecializedInterfaceType(type);
 
           return SpecializedGenericEnv->mapTypeOutOfContext(
```

Comment:
The map out/into context should not be necessary. In fact, this can just be SpecializedSubstFnTy = CalleeFnTy->substGenericArgs(M, SpecializedInterfaceToCallerArchetypeMap)